packages/databricks-vscode/src/ui/configuration-view/ConfigurationTreeViewManager.ts

```diff
@@ -0,0 +1,38 @@
+import {
+    TreeView,
+    TreeCheckboxChangeEvent,
+    Disposable,
+    TreeItemCheckboxState,
+} from "vscode";
+import {ConfigurationTreeItem} from "./types";
+import {ConfigModel} from "../../configuration/models/ConfigModel";
+import {CLUSTER_OVERRIDE_CHECKBOX_ID} from "./ClusterComponent";
+
+export class ConfigurationTreeViewManager implements Disposable {
+    private readonly disposables: Disposable[] = [];
+    constructor(
+        readonly treeView: TreeView<ConfigurationTreeItem>,
+        readonly configModel: ConfigModel
+    ) {
+        this.disposables.push(
+            treeView.onDidChangeCheckboxState(
+                async (e: TreeCheckboxChangeEvent<ConfigurationTreeItem>) => {
+                    await Promise.all(
+                        e.items.map(async ([item, state]) => {
+                            if (item.id === CLUSTER_OVERRIDE_CHECKBOX_ID) {
+                                await this.configModel.set(
```

Comment:
Why can't we do it in the ClusterComponent directly and not rely on treeView-level change events? E.g. execute a command each time the checkbox is changed

Reply:
The only way to listen to changes to builtin check box is through tree view. We will have to implement our own check boxes otherwise.